### PR TITLE
Feature/goals sublist

### LINKED ISSUE #423: Redirect user to sublist page
// TODO: Add description

### LINKED ISSUE #424: Make the options on my goal items working
// TODO: Add description

### PR DESCRIPTION
**Pull request by pulkitvyas08**
Resolves #424 
Resolves #423 

The task breakdown of this feature could have been much better. This PR's work could have been broken down into at least 4 more issues, there was the extension to DB schema and API involved, UI additions, pages, and router handling, extending existing components and pages, and some validation too.

I have changed the relationship between goal-subgoal to a symbolic one instead of the contained way done earlier. This will allow for better storage efficiency and with some clever code for defining this relationship and putting some validation, we would be able to extend this to the DAG representation we want.

### CODE REVIEW

**Review by tijlleenders — DISMISSED** [review]
@pulkitvyas08 This PR is amazing!

We've reached another milestone.
Thr title of the issue is short - but is - like you say - packed with multiple underlyimg issues.

Great stuff to demo today!

File: src/components/GoalsComponents/GoalsForm.tsx [hunk 1]
```diff
@@ -78,15 +83,28 @@ export const GoalsForm = ({ selectedColorIndex }: { selectedColorIndex: number }
...
+      Number(goalDuration.split(" ")[0]),
+      null,
+      null,
+      0,
+      parentGoalId!
+    );
+    const newGoalId = await addGoal(newGoal);
+    if (parentGoalId) {
+      const parentGoal = await getGoal(parentGoalId);
+      const newSublist = parentGoal?.sublist === undefined ? [newGoalId] : [...parentGoal?.sublist, newGoalId];
```
Comment (Tushar-4781) [inline src/components/GoalsComponents/GoalsForm.tsx:98]: you already checked whether the parentGoal.sublist is defined or not so while executing the block why use the '?' . I think that's why it is failing the tc

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]
I think i found the issue

File: src/components/GoalsComponents/GoalsForm.tsx (hunk 1 above)
Comment (Tushar-4781) [reply src/components/GoalsComponents/GoalsForm.tsx:98]: not holding this PR. Quite amount of commits I c. I m Pushing the requested changes

**Review by Tushar-4781 — APPROVED** [review]

File: src/components/GoalsComponents/GoalsForm.tsx (hunk 1 above)
Comment (pulkitvyas08) [reply src/components/GoalsComponents/GoalsForm.tsx]: Right, that makes so much sense. I was using optional chaining earlier without the conditional statement so it was working, but now that isn't really needed, thanks for the commits

**Pull request merged**